### ISSUE TITLE
Working with sessions from MockHttpServletRequest

### ISSUE THREAD

**Issue by mthmulders** [mockrunner/mockrunner#55]
As I was writing some tests involving `HttpServlet` and `HttpSession`, I noticed the following:
1. When you invoke `MockHttpServletRequest#getSession(false)` after having set a `MockHttpSession`, it will still return `null`.
1. When you invoke `MockHttpServletRequest#getSession(true)` (without further interaction with that `HttpServletRequest`) it will return `null`.

I would like to suggest the following changes:
1. When you invoke `MockHttpServletRequest#getSession(false)` after having set a `MockHttpSession`, it will return the previously set `MockHttpSession`.
1. When you invoke `MockHttpServletRequest#getSession(true)` (without further interaction with that `HttpServletRequest`) it will return a new instance of `MockHttpSession`, _unless_ there already is an instance of `MockHttpSession`.

These changes would make the `MockHttpServletRequest` a bit more conformant to the Servlet spec.
How would you feel about such changes? I can contribute a PR if the above suggestions would be approved.

**Comment by cemartins** [comment]
I agree with you. Would be great if you could provide a pull request with those changes.
